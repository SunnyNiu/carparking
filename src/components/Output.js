import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

const Output = ({ output }) => {
  return (
    <div>
      <label>Report:</label>
      <label>{output}</label>
    </div>
  );
};

Output.propTypes = {
  output: PropTypes.string.isRequired,
};
const mapStateToProps = state => {
  return {
    output: state.app.output,
  };
};
export default connect(mapStateToProps)(Output);
