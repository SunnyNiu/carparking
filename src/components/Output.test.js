import React from 'react';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import Output from './Output';

const mockStore = configureStore([]);

describe('<Output/> component', () => {
  const store = mockStore({
    app: {
      location: {
        x: 0,
        y: 4,
        facing: 'NORTH',
      },
      output: '0,4,NORTH',
    },
  });

  const wrapper = mount(
    <Provider store={store}>
      <Output />
    </Provider>
  );
  it('Output snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('show all output', () => {
    expect(
      wrapper.containsAllMatchingElements([<label>0,4,NORTH</label>])
    ).toBe(true);
  });
});
