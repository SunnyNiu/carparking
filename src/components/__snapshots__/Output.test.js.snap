// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Output/> component Output snapshot 1`] = `
<Provider
  store={
    Object {
      "clearActions": [Function],
      "dispatch": [Function],
      "getActions": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
    }
  }
>
  <Connect(Output)>
    <Output
      dispatch={[Function]}
      output="0,4,NORTH"
    >
      <div>
        <label>
          Report:
        </label>
        <label>
          0,4,NORTH
        </label>
      </div>
    </Output>
  </Connect(Output)>
</Provider>
`;
